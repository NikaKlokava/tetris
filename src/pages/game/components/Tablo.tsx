import cl from "./styles.module.css";
import { memo } from "react";

type Props = {
  rows: number;
  level: number;
  score: number;
  gameOver: boolean;
};

export const Tablo = memo(({ rows, level, score, gameOver }: Props) => {
  return (
    <div className={cl.tablo}>
      {gameOver ? (
        <div className={cl.game_over}>Game Over!</div>
      ) : (
        <>
          <div className={cl.tablo_item}>{`Score : ${score}`}</div>
          <div className={cl.tablo_item}>{`Rows : ${rows}`}</div>
          <div className={cl.tablo_item}>{`Level : ${level}`}</div>
        </>
      )}
    </div>
  );
});
