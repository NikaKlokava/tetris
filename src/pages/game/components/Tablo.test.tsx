import renderer from "react-test-renderer";
import { Tablo } from "./Tablo";

describe("Test the Tablo component", () => {
  test("The Tablo renders correctly", () => {
    const tabloSnap = renderer
      .create(<Tablo rows={0} level={0} score={0} gameOver={false} />)
      .toJSON();
    expect(tabloSnap).toMatchSnapshot();
  });
});
