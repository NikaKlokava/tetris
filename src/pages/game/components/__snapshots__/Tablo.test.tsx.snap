// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Tablo component The Tablo renders correctly 1`] = `
<div
  className="tablo"
>
  <div
    className="tablo_item"
  >
    Score : 0
  </div>
  <div
    className="tablo_item"
  >
    Rows : 0
  </div>
  <div
    className="tablo_item"
  >
    Level : 0
  </div>
</div>
`;
